From 9762e29881b362b71c58095905c3472676d20f3f Mon Sep 17 00:00:00 2001
From: Anson Huang <b20788@freescale.com>
Date: Tue, 7 Apr 2015 18:20:28 +0800
Subject: [PATCH] MLK-10595-1 ARM: imx: correct gpc a7_bsc wakeup source
 setting

GPC_LPCR_A7_BSC should be set to be waked up by GIC/GPC both,
the value should be 0x3, otherwise, first time DSM will be
entered by mistake when program the GPC low power mode register
but ARM NOT enter wfi yet.

Signed-off-by: Anson Huang <b20788@freescale.com>
---
 arch/arm/mach-imx/gpcv2.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/arch/arm/mach-imx/gpcv2.c b/arch/arm/mach-imx/gpcv2.c
index 198e063..7c4691e 100644
--- a/arch/arm/mach-imx/gpcv2.c
+++ b/arch/arm/mach-imx/gpcv2.c
@@ -512,7 +512,7 @@ void __init imx_gpcv2_init(void)
 	}
 
 	/* only external IRQs to wake up LPM and core 0/1 */
-	writel_relaxed(BM_LPCR_A7_BSC_IRQ_SRC_A7_WAKEUP,
+	writel_relaxed(0x3 << BP_LPCR_A7_BSC_IRQ_SRC,
 		gpc_base + GPC_LPCR_A7_BSC);
 	/* mask m4 dsm trigger */
 	writel_relaxed(readl_relaxed(gpc_base + GPC_LPCR_M4) |
-- 
1.8.0

